components/van-ui.ts

```diff
@@ -499,3 +499,188 @@ export const Banner = (
+  readonly closed: State<boolean>
+  readonly x?: State<number>
+  readonly y?: State<number>
+  readonly width?: State<number>
+  readonly height?: State<number>
+  readonly windowStyleOverrides?: CSSPropertyBag
+  readonly headerStyleOverrides?: CSSPropertyBag
+  readonly childrenContainerStyleOverrides?: CSSPropertyBag
+  readonly closeCross?: boolean
+}
+
+export const FloatingWindow = (
+  {
+    title,
+    closed,
+    x = van.state(100),
+    y = van.state(100),
+    width = van.state(300),
+    height = van.state(200),
+    windowStyleOverrides = {},
+    headerStyleOverrides = {},
+    childrenContainerStyleOverrides = {},
+    closeCross = false
+  }: FloatingWindowProps,
+  ...children: readonly ChildDom[]
+) => {
+  let dragging = van.state(false)
+  let resizingDirection = van.state<string | null>(null)
+  let startX = van.state(0)
+  let startY = van.state(0)
+  let startWidth = van.state(0)
+  let startHeight = van.state(0)
+
+  const onMouseDown = (e: MouseEvent) => {
+    dragging.val = true
+    startX.val = e.clientX
+    startY.val = e.clientY
+    document.body.style.userSelect = 'none'
+  }
+
+  const onResizeMouseDown = (direction: string) => (e: MouseEvent) => {
+    resizingDirection.val = direction
+    startX.val = e.clientX
+    startY.val = e.clientY
+    startWidth.val = width.val
+    startHeight.val = height.val
+    document.body.style.userSelect = 'none'
+  }
+
+  const onMouseMove = (e: MouseEvent) => {
+    if (dragging.val) {
+      x.val += e.clientX - startX.val
+      y.val += e.clientY - startY.val
+      startX.val = e.clientX
+      startY.val = e.clientY
+    } else if (resizingDirection.val) {
+      const deltaX = e.clientX - startX.val
+      const deltaY = e.clientY - startY.val
+
+      if (resizingDirection.val.includes('right')) {
+        width.val = startWidth.val + deltaX
+      }
+      if (resizingDirection.val.includes('bottom')) {
+        height.val = startHeight.val + deltaY
+      }
+    }
+  }
+
+
+  const onMouseUp = () => {
+    dragging.val = false
+    resizingDirection.val = null
+    document.body.style.userSelect = ''
+  }
+
+  document.addEventListener('mousemove', onMouseMove)
+  document.addEventListener('mouseup', onMouseUp)
+
+  const grabAreaBgColor = 'transparent';
+
+  return () => closed.val ? null : van.add(
+    div({
+      style: toStyleStr({
+        position: 'fixed',
+        left: `${x.val}px`,
+        top: `${y.val}px`,
+        width: `${width.val}px`,
+        height: `${height.val}px`,
+        'background-color': 'white',
+        border: '1px solid black',
+        'border-radius': '0.5rem',
+        overflow: 'hidden',
+        ...windowStyleOverrides,
+      }),
+    },
+      div({
+        style: toStyleStr({
+          cursor: 'move',
+          position: 'absolute',
+          left: 0,
+          top: 0,
+          width: '100%',
+          height: "1rem"
+        }),
+        onmousedown: onMouseDown,
+      }),
```

Comment:
Actually, I don't quite understand the purpose of this `<div>` element. When I tried to remove the element, nothing seems to be effected. Is it redundant or there is something you intended to do?

Reply:
I we remove this div this window is not draggable when your title bar is hidden.
if remove "Window with Tabs with custom close button" can be drag. its useless most of the time, if you want we can add a boolean titlebar? and when its false we add this div to allow drag.

Reply:
Can we show this `<div>` only when `!title` is true, and show the `<header>` below only when `title` is true?